builtin/readline_osh.py

```diff
@@ -376,3 +376,94 @@ def Run(self, cmd_val):
             item = readline.get_history_item(i)
             self.f.write('%5d  %s\n' % (i, item))
         return 0
+
+class Fc(vm._Builtin):
+    """Show or execute commands from interactive command history."""
+
+    def __init__(
+            self,
+            readline,  # type: Optional[Readline]
+            f,  # type: mylib.Writer
+    ):
+        # type: (...) -> None
+        self.readline = readline
+        self.f = f  # this hook is for unit testing only
+
+    def Run(self, cmd_val):
+        # type: (cmd_value.Argv) -> int
+        readline = self.readline
+        if not readline:
+            e_usage("is disabled because Oils wasn't compiled with 'readline'",
+                    loc.Missing)
+
+        attrs, arg_r = flag_util.ParseCmdVal('fc', cmd_val)
+        arg = arg_types.fc(attrs.attrs)
+
+        # Returns 0 items in non-interactive mode?
+        num_items = readline.get_current_history_length()
+
+        first_arg, first_arg_loc = arg_r.Peek2()
+        if first_arg is None:
+            if num_items > 16:
```

Comment:
Hm, where does the number 16 come from?  Is that what bash does?

May be worth a comment